core/src/stored/ndmp_tape.cc

```diff
@@ -398,40 +398,51 @@ static inline bool bndmp_read_data_from_block(JobControlRecord* jcr,
     // See if we are processing some sort of label?
     if (rctx->rec->FileIndex < 0) { continue; }
 
-    /*
-     * Here we should have read a record from the block which contains some
-     * data. Its either:
-     *
-     * - STREAM_UNIX_ATTRIBUTES
-     *      Which is the start of the dump when we encounter that we just read
-     * the next record.
-     * - STREAM_FILE_DATA
-     *      Normal NDMP data.
-     * - STREAM_NDMP_SEPARATOR
-     *      End of NDMP data stream.
-     *
-     * anything other means a corrupted stream of records and means we give an
-     * EOF.
-     */
-    switch (rctx->rec->maskedStream) {
-      case STREAM_UNIX_ATTRIBUTES:
+    DeviceRecord* rec = rctx->rec;
+    // Perform record translations only if data is compressed
+    // as NDMP needs to be decompressed in any case
+    if (rctx->rec->maskedStream == STREAM_COMPRESSED_DATA) {
```

Comment:
When limiting the GeneratePluginEvent() to STREAM_COMPRESSED_DATE, then it will work for autoxflate, but would probably fail (or simply not do the right thing) with other potential plugins.